classes/image_populate_mfgtool.bbclass

```diff
@@ -0,0 +1,142 @@
+# determine the behavior of each script by providing up to two order-dependent arguments, which are
+# separated by commas. You can omit any argument you like but must retain the separating commas. The
+# order is important and specifies the following:
+#
+#   1. Extra dependencies that should be added to the do_populate_mfgtool task, if the script is
+#      enabled.
+#   2. Extra binaries that should be added to the bundle, if the script is enabled.
+#
+# For example:
+#
+# ,----[ Defining bootloader.uuu.in script ]
+# | MFGTOOLSCRIPT = "bootloader.uuu.in"
+# | MFGTOOLSCRIPT[bootloader.uuu.in] = "dep1:do_deploy dep2:do_deploy,file1 file2"
+# `----
+#
+# The virtual/bootloader:do_deploy is added to do_populate_mfgtool dependencies and ${UBOOT_BINARY}
+# copied to the bundle, only if the script is enabled.
+#
+# During the mfgtool bundle generation, the uuu.in files are processed and some variables
+# replaced. The variables are:
+#
+#   - MACHINE
+#   - UBOOT_BINARY
+#   - SPL_BINARY
+#   - IMAGE_BASENAME
+#
+# Copyright 2022-2023 (C) O.S. Systems Software LTDA.
+#
+# SPDX-License-Identifier: MIT
+
+MFGTOOL_FILESPATH ??= " \
+    ${@base_set_filespath(["%s/mfgtool" % p for p in "${BBPATH}".split(":")] \
+                             + ["${FILE_DIRNAME}/${BP}/mfgtool", \
+                                "${FILE_DIRNAME}/${BPN}/mfgtool", \
+                                "${FILE_DIRNAME}/files/mfgtool"] \
+                          , d)} \
+"
+
+MFGTOOLDIR = "${WORKDIR}/mfgtool-${PN}"
+do_populate_mfgtool[dirs] = "${MFGTOOLDIR}"
+do_populate_mfgtool[cleandirs] = "${MFGTOOLDIR}"
+
+addtask populate_mfgtool after do_image_complete do_unpack before do_deploy
+do_populate_mfgtool[dirs] ?= "${DEPLOY_DIR_IMAGE} ${WORKDIR}"
+do_populate_mfgtool[nostamp] = "1"
+do_populate_mfgtool[recrdeptask] += "do_deploy"
+do_populate_mfgtool[depends] += "uuu-bin-native:do_populate_sysroot"
+
+python () {
+    # Process MFGTOOLSCRIPT and handle its fields.
+    mfgtoolscript_flags = d.getVarFlags('MFGTOOLSCRIPT') or {}
+    mfgtoolscript_depends = []
+    mfgtoolscript_deploy_files = []
+    if mfgtoolscript_flags:
+        mfgtoolscript = (d.getVar('MFGTOOLSCRIPT') or "").split()
+        pn = d.getVar("PN")
+
+        for flag, flagval in sorted(mfgtoolscript_flags.items()):
+            items = flagval.split(",")
+            num = len(items)
+            if num > 2:
+                bb.error("%s: MFGTOOLSCRIPT[%s] Only \"depends,deploy files\" can be specified!" % (pn, flag))
+
+            if flag in mfgtoolscript:
+                if num >= 2 and items[1]:
+                    mfgtoolscript_deploy_files.append(items[1])
+                if num >= 1 and items[0]:
+                    mfgtoolscript_depends.append(items[0])
+
+        d.appendVarFlag('do_populate_mfgtool', 'depends', ' ' + ' '.join(mfgtoolscript_depends))
+
+    d.setVar('MFGTOOLSCRIPT_DEPLOY_FILES', ' '.join(mfgtoolscript_deploy_files))
```

Comment:
as `MFGTOOLSCRIPT_DEPLOY_FILES` is an internal variable of the class, something like `_MFGTOOLSCRIPT_DEPLOY_FILES` can be better.

Reply:
Fixed.